pkg/services/ngalert/models/instance_test.go

```diff
@@ -53,41 +53,225 @@ func buildTestInstanceStateTypeIsValidName(instanceType InstanceStateType, expec
 	return fmt.Sprintf("%q should not be valid", instanceType)
 }
 
+func TestInstanceCauseType_IsValid(t *testing.T) {
+	testCases := []struct {
+		instanceCauseType InstanceCauseType
+		expectedValidity  bool
+	}{
+		{
+			instanceCauseType: InstanceNoCause,
+			expectedValidity:  true,
+		},
+		{
+			instanceCauseType: InstanceCauseFiring,
+			expectedValidity:  true,
+		},
+		{
+			instanceCauseType: InstanceCauseError,
+			expectedValidity:  true,
+		},
+		{
+			instanceCauseType: InstanceCauseNoData,
+			expectedValidity:  true,
+		},
+		{
+			instanceCauseType: InstanceCauseType("notAValidInstancePendingStateType"),
+			expectedValidity:  false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(buildTestInstancePendingStateTypeIsValidName(tc.instanceCauseType, tc.expectedValidity), func(t *testing.T) {
```

Comment:
```suggestion
		t.Run(tc.name func(t *testing.T) {
```

It's also idiomatic to just have the test name in the structs. You don't need a function here, it simplifies the tests, and it also allows for more descriptive test names.